spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/archaius/ArchaiusAutoConfiguration.java

```diff
@@ -165,14 +158,30 @@ protected void configureArchaius(ConfigurableEnvironmentConfiguration envConfig)
 			config.setContainerConfigurationIndex(config
 					.getIndexOfConfiguration(appOverrideConfig));
 
-			ConfigurationManager.install(config);
-		}
-		else {
+			addArchaiusConfiguration(config);
+		} else {
 			// TODO: reinstall ConfigurationManager
 			log.warn("Netflix ConfigurationManager has already been installed, unable to re-install");
 		}
 	}
 
+	private void addArchaiusConfiguration(ConcurrentCompositeConfiguration config) {
```

Comment:
I see, I will look over it. As I see I don't think there will be a way to restrict the use of static archaius methods, but the collection of injected configuration beans should work also.